src/python/ImpactX.cpp

```diff
@@ -29,6 +30,34 @@ namespace impactx {
     struct Config {};
 }
 
+namespace detail
+{
+    /** Helper Function for Property Getters
+     *
+     * This queries an amrex::ParmParse entry. This throws a
+     * std::runtime_error if the entry is not found.
+     *
+     * This handles most common throw exception logic in ImpactX instead of
+     * going over library boundaries via amrex::Abort().
+     *
+     * @tparam T type of the amrex::ParmParse entry
+     * @param prefix the prefix, e.g., "impactx" or "amr"
+     * @param name the actual key of the entry, e.g., "particle_shape"
+     * @return the queried value (or throws if not found)
+     */
+    template< typename T>
+    auto get_or_throw(std::string const prefix, std::string const name)
```

Comment:
We can check in the future if we find a reasonable location to set default parameters without setting them multiple times throughout the code.

Maybe we need a `ReadParameters`-like function called `set_default_parameters` after all... :-S